kernel/gic/src/gic/redist_interface.rs

```diff
@@ -44,18 +57,17 @@ const TIMEOUT_ITERATIONS: usize = 0xffff;
 /// Initializes the redistributor by waking
 /// it up and checking that it's awake
 pub fn init(registers: &mut GicRegisters) -> Result<(), &'static str> {
-    let mut reg;
-    reg = registers.read_volatile(offset::RD_WAKER);
+    let mut reg = registers.read_volatile(offset::WAKER);
 
     // Wake the redistributor
-    reg &= !RD_WAKER_PROCESSOR_SLEEP;
+    reg &= !WAKER_PROCESSOR_SLEEP;
 
-    registers.write_volatile(offset::RD_WAKER, reg);
+    registers.write_volatile(offset::WAKER, reg);
 
     // then poll ChildrenAsleep until it's cleared
 
     let children_asleep = || {
-        registers.read_volatile(offset::RD_WAKER) & RD_WAKER_CHLIDREN_ASLEEP > 0
+        registers.read_volatile(offset::WAKER) & WAKER_CHLIDREN_ASLEEP > 0
     };
 
     let mut counter = 0;
```

Comment:
If this closure based loop is here so it can't get optimized out: I think it still can: https://godbolt.org/z/9WK56qdqr

Reply:
huh? 

I'm not sure what this comment means. @NathanRoyer?

Reply:
Afaik the rust compiler can optimize timeout loops away unless there is some operation inside the loop which can't be reasoned about by the compiler, i.e. volatile accesses or an `asm!` block. I guess in this case it optimizes the `timeout` closure (on line 74, not shown in this comment window) in a way that the volatile read happens inside of it.

Reply:
Yeah, if rustc optimized a `read_volatile` away it would be a bug imo. Anyway, this actually compiles to a volatile read at the moment.

Reply:
Ah, I see, github hid the relevant code -- @hecatia-elegua was talking about the timeout checking loop being optimized away, not the volatile register read. They're right, I checked the actual asm and it wasn't in the loop body. I've changed it to fix that.

Reply:
Wait I thought it was 🤣 
well whatever, if it works for you